localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/state_task_lambda.py

```diff
@@ -143,6 +148,10 @@ def _eval_execution(self, env: Environment) -> None:
             self.timeout.eval(env=env)
             timeout_seconds = env.stack.pop()
             scheduled_event_details["timeoutInSeconds"] = timeout_seconds
+        if self.credentials:
```

Comment:
nit: Not sure if they're ever interchangable but would be curious if we'd ever want to use the stored creds in `self.credentials` as the `TaskCredentials`.

Reply:
The Credentials type carries the list of definitions on how to compute the credentials, whereas the TaskCredentials carries the computes credential fields